streams/src/main/java/org/apache/kafka/streams/state/internals/GlobalStateStoreProvider.java

```diff
@@ -41,6 +44,11 @@ public GlobalStateStoreProvider(final Map<String, StateStore> globalStateStores)
         if (!store.isOpen()) {
             throw new InvalidStateStoreException("the state store, " + storeName + ", is not open.");
         }
+        if (store instanceof TimestampedKeyValueStore && queryableStoreType instanceof QueryableStoreTypes.KeyValueStoreType) {
+            return (List<T>) Collections.singletonList(new ReadOnlyKeyValueStoreFacade((TimestampedKeyValueStore<Object, Object>) store));
+        } else if (store instanceof TimestampedWindowStore && queryableStoreType instanceof QueryableStoreTypes.WindowStoreType) {
+            return (List<T>) Collections.singletonList(new ReadOnlyWindowStoreFacade((TimestampedWindowStore<Object, Object>) store));
+        }
```

Comment:
Note, change is actually part of #6661 and just added here to make the tests pass -- #6661 should be merged first.